src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -356,5 +505,30 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
 
             _postElement?.WriteTo(writer, encoder);
         }
+
+        private class AttributeContent : IHtmlContent
+        {
+            public AttributeContent(IHtmlContent inner)
+            {
+                Inner = inner;
+            }
+
+            public IHtmlContent Inner { get; }
+
+            public void WriteTo(TextWriter writer, HtmlEncoder encoder)
+            {
+                // There's no way of tracking the attribute value quotations in the Razor source. Therefore, we
+                // must escape any IHtmlContent double quote values in the case that a user wrote:
+                // <p name='A " is valid in single quotes'></p>
+                using (var stringWriter = new StringWriter())
```

Comment:
Suggest allocating the `StringWriter()` in the containing class (lazily), passing into the constructor, and clearing on every `WriteTo()` call. That should somewhat reduce the explosion of `char[]`, `StringBuilder` and `StringWriter` instances.